src/OrchardCore/OrchardCore/Shell/Builders/ShellContainerFactory.cs

```diff
@@ -123,7 +123,9 @@ public async Task<IServiceProvider> CreateContainerAsync(ShellSettings settings,
         // Let any module add custom service descriptors to the tenant.
         foreach (var startup in startups)
         {
-            var feature = blueprint.Dependencies.FirstOrDefault(x => x.Key == startup.GetType()).Value?.FirstOrDefault();
+            // For raw startup classes, use the original type named 'Startup' to find the feature.
+            var startupType = startup is StartupBaseMock mock ? mock.StartupType : startup.GetType();
```

Comment:
StartupBaseMock in the main project seems very strange

Reply:
That is an existing internal class which gets added to DI as a substitute for raw startup classes. ***Raw*** meaning public classes named `Startup`, which are not inheriting from `IStartup` or its default implementation `Startupbase`.